File: Java/src/main/java/br/com/locaweb/locamail/api/model/Alteracao.java
package br.com.locaweb.locamail.api.model;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "T_LCW_ALTERACAO")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode
public class Alteracao {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_alteracao")
    private Long id_alteracao;

    @Column(name = "alt_id_usuario", nullable = false)
    private Long alt_id_usuario;

    @Column(name = "alt_id_email", nullable = false)
    private Long alt_id_email;

    @Column(name = "id_pasta")
    private Long id_pasta;

    @Column(name = "importante", nullable = false)
    private Boolean importante;

    @Column(name = "lido", nullable = false)
    private Boolean lido;

    @Column(name = "arquivado", nullable = false)
    private Boolean arquivado;

    @Column(name = "excluido", nullable = false)
    private Boolean excluido;

    @Column(name = "spam", nullable = false)
    private Boolean spam;
}

File: Kotlin/app/src/main/java/br/com/fiap/locawebmailapp/model/Alteracao.kt
package br.com.fiap.locawebmailapp.model

import androidx.annotation.Nullable
import androidx.compose.runtime.Immutable
import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey

@Immutable
@Entity(tableName = "T_LCW_ALTERACAO",
    foreignKeys =
    [ForeignKey(
        entity = Usuario::class,
        childColumns = ["alt_id_usuario"],
        parentColumns = ["id_usuario"]
    ),
        ForeignKey(
            entity = Email::class,
            childColumns = ["alt_id_email"],
            parentColumns = ["id_email"]
        ),
        ForeignKey(
            entity = Pasta::class,
            childColumns = ["id_pasta"],
            parentColumns = ["id_pasta"]
        )
    ])
data class Alteracao(
    @PrimaryKey(autoGenerate = true) var id_alteracao: Long = 0,
    var alt_id_usuario: Long = 1,
    var alt_id_email: Long = 1,
    var id_pasta: Long? = null,
    var importante: Boolean = false,
    var lido: Boolean = false,
    var arquivado: Boolean = false,
    var excluido: Boolean = false,
    var spam: Boolean = false

)